src/pfe/file-watcher/server/src/extensions/IExtensionProject.ts

```diff
@@ -130,7 +130,27 @@ export interface IExtensionProject {
      * @param projectID <Required | string> - An alphanumeric identifier for a project.
      * @param projectLocation <Required | string> - The project location directory.
      *
-     * @returns string
+     * @returns Promise<string>
+     */
+    getContainerName?(projectID: string, projectLocation: string): Promise<string>;
+
+    /**
+     * @function
+     * @description Return a reproducible app name for the project.
+     *
+     * @param projectID <Required | string> - An alphanumeric identifier for a project.
+     *
+     * @returns Promise<string>
+     */
+    getAppName?(projectID: string): Promise<string>;
+
+    /**
+     * @function
+     * @description Get the internal app port that the project uses.
+     *
+     * @param projectID <Required | string> - An alphanumeric identifier for a project.
+     *
+     * @returns Promise<string>
      */
-    getContainerName?(projectID: string, projectLocation: string): string;
+    getAppPort?(projectID: string): Promise<string>;
```

Comment:
We are adding these functions. Does it mean that the Appsody needs to react to implement these as well to avoid breakage?

Reply:
Nope, we put `?` mark for the function, that means the function in the interface is optional.